### ISSUE TITLE
Failed to fetch tags for private Github repo.

### ISSUE THREAD

**Issue by robocoder99** [jenkins-x/jx#1180]
I have a go project setup on a kube cluster on GKE. When I create a pull request to the master branch on my project repo jenkins successfully builds and deploys the preview environment but when I merge the PR it fails to build and promote to staging. I cannot find any documentation on how to handle this situation. Here are my logs. I would appreciate any help or insight.

```[peerintuit] Running shell script
+ make tag
sed -i -e "s/version:.*/version: failed to get new version error fetching tags: %vexit status 128/" Chart.yaml
sed -i -e "s/repository: .*/repository: 10.23.242.210:5000\/robocoder99\/peerintuit/" values.yaml
sed -i -e "s/tag: .*/tag: failed to get new version error fetching tags: %vexit status 128/" values.yaml
git add --all
git commit -m "release failed to get new version error fetching tags: %vexit status 128" --allow-empty # if first release then no verion update is performed
[master 5cdd155] release failed to get new version error fetching tags: %vexit status 128
 5 files changed, 11 insertions(+), 3 deletions(-)
 create mode 100644 VERSION
 create mode 100644 charts/peerintuit@tmp/durable-0b3a905c/jenkins-log.txt
 create mode 100755 charts/peerintuit@tmp/durable-0b3a905c/script.sh
git tag -fa vfailed to get new version error fetching tags: %vexit status 128 -m "Release version failed to get new version error fetching tags: %vexit status 128"
fatal: too many params
make: *** [tag] Error 128
```

EDIT: I have confirmed this is functional for public github repos. It seems to be an issue with authentication.

**Comment by robocoder99** [comment]
I have fixed the issue locally. The Jenkinsfile is not passing the GITHUB_AUTH_TOKEN environmental variable to jx-release-version and is doing so before jx step git credentials is executed either. This results in jx-release-version falling back to an unauthenticated git tag request which in turn fails for private repositories. I reordered the Jenkinsfile slightly and modified the line dispatching jx-release-version to also retrieve and pass in the auth token. My piplines are now building fine. Upon investigating it looks like the issue lies within https://github.com/jenkins-x/draft-packs/blob/master/packs/go/Jenkinsfile specifically for my case but the other packs would need to be modified as well.
